dex/fiatrates/sources.go

```diff
@@ -0,0 +1,246 @@
+package fiatrates
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"decred.org/dcrdex/dex"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const (
+	defaultRefreshInterval = 5 * time.Minute
+	messariRefreshInterval = 10 * time.Minute
+
+	cryptoCompare              = "CryptoCompare"
+	cryptoComparePriceEndpoint = "https://min-api.cryptocompare.com/data/pricemulti?fsyms=%s&tsyms=USD"
+
+	binance                = "Binance"
+	binancePriceEndpoint   = "https://api3.binance.com/api/v3/ticker/price?symbols=[%s]"
+	binanceUSPriceEndpoint = "https://api.binance.us/api/v3/ticker/price?symbols=[%s]"
+
+	coinpaprika              = "Coinparika"
+	coinpaprikaPriceEndpoint = "https://api.coinpaprika.com/v1/tickers"
+
+	messari              = "Messari"
+	messariPriceEndpoint = "https://data.messari.io/api/v1/assets/%s/metrics/market-data"
+
+	kuCoin              = "KuCoin"
+	kuCoinPriceEndpoint = "https://api.kucoin.com/api/v1/prices?currencies=%s"
+)
+
+var (
+	upperCaser = cases.Upper(language.AmericanEnglish)
+)
+
+func fiatSources(cfg Config) []*source {
+	disabledSources := strings.ToLower(cfg.DisabledFiatSources)
+	sources := []*source{
+		{
+			name:            cryptoCompare,
+			requestInterval: defaultRefreshInterval,
+			disabled:        cfg.CryptoCompareAPIKey == "" || strings.Contains(disabledSources, strings.ToLower(cryptoCompare)),
+			getRates: func(ctx context.Context, assets []string, _ dex.Logger) (map[string]float64, error) {
```

Comment:
We have terms for various asset representations. An "asset ID" and a "symbol" are the unique identifiers for assets defined in **dex/bip-id.go**. Symbols are always lower-case and use only ascii characters. We have "conventional unit", or I prefer "ticker", which are defined as part of an assets `UnitInfo`. These are always upper-case and are probably the best candidate for API compatibility. Regardless, nothing you can pass as a `string` should be called an "asset". That language is more for like `dex.Asset` or a `core.SupportedAsset`.